### PR TITLE
[Yaml] deprecate the !str tag

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        | 

The tag specified in the YAML spec is actually !!str.

### CODE REVIEW

**Review by ogizanagi — APPROVED** [review]

File: src/Symfony/Component/Yaml/Inline.php [hunk 1]
```diff
@@ -609,7 +609,11 @@ private static function evaluateScalar($scalar, $flags, $references = array())
             case $scalar[0] === '!':
                 switch (true) {
                     case 0 === strpos($scalar, '!str'):
+                        @trigger_error('Support for the !str tag is deprecated since version 3.4. Use the !!str tag instead.', E_USER_DEPRECATED);
+
                         return (string) substr($scalar, 5);
+                    case 0 === strpos($scalar, '!!str '):
+                        return (string) substr($scalar, 6);
```
Comment (GuilhemN) [inline src/Symfony/Component/Yaml/Inline.php]: While doing this, can't we use `parseScalar` here?

**Review by javiereguiluz — APPROVED** [review]

File: src/Symfony/Component/Yaml/Inline.php (hunk 1 above)
Comment (javiereguiluz) [inline src/Symfony/Component/Yaml/Inline.php]: Maybe it's better to do that in a separate PR to not mix too many things in the same PR.

Comment (GuilhemN) [inline src/Symfony/Component/Yaml/Inline.php]: Ok, fine for me :)

**Review by GuilhemN — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

File: src/Symfony/Component/Yaml/Inline.php (hunk 1 above)
Comment (stof) [inline src/Symfony/Component/Yaml/Inline.php]: if we use parseScalar, we may end up parsing the value as a YAML timestamp for instance, which will break the usage of the tag (see `not-date: !!str 2002-04-28` in the tests for instance)

Comment (GuilhemN) [inline src/Symfony/Component/Yaml/Inline.php]: Fortunately `parseScalar` has a `$evaluate` argument :)
